File: scala/p2.scala
// Solution for Project Euler problem 2
// http://projecteuler.net/index.php?section=problems&id=2

def fib(x: List[Int], y: Int): List[Int] = x match {
  case Nil => error("")
  case i :: Nil => error("")
  case i :: j :: xs => if (i + j < y) fib(i + j :: x, y) else x
}

println(fib(List(1,1), 4000000).reduceLeft((a, b) =>
  if ( b % 2 == 0) a+b else a))

File: python/p2.py
# Solution for Project Euler problem 2
# http://projecteuler.net/index.php?section=problems&id=2

def fib(limit):
    i = [1, 1]
    while i[-1] + i[-2] < limit:
        i.append(i[-1] + i[-2])
    return i


if __name__ == '__main__':
    print sum([x for x in fib(4000000) if x % 2 == 0])
